google-cloud-firestore/lib/google/cloud/firestore/database.rb

```diff
@@ -0,0 +1,1013 @@
+        # @example Create a document using a document reference:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a document reference
+        #   nyc_ref = firestore.doc "cities/NYC"
+        #
+        #   # Create a document
+        #   firestore.create(nyc_ref, { name: "New York City" })
+        #
+        def create doc, data
+          batch { |b| b.create doc, data }
+        end
+
+        ##
+        # Write to document with the provided object values. If the document
+        # does not exist, it will be created. By default, the provided data
+        # overwrites existing data, but the provided data can be merged into the
+        # existing document using the `merge` argument.
+        #
+        # If you're not sure whether the document exists, pass the option to
+        # merge the new data with any existing document to avoid overwriting
+        # entire documents.
+        #
+        # @param [String, Document::Reference] doc A string representing the
+        #   path of the document, or a document reference object.
+        # @param [Hash] data The document's fields and values.
+        # @param [true, String|Symbol, Array<String|Symbol>] merge When provided
+        #   and `true` all data is merged with the existing document data. When
+        #   provided only the specified as a list of field paths are merged with
+        #   the existing document data. The default is to overwrite the existing
+        #   document data.
+        #
+        # @return [Time] The time the changes were committed
+        #
+        # @example Set a document using a document path:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Set a document
+        #   firestore.set("cities/NYC", { name: "New York City" })
+        #
+        # @example Create a document using a document reference:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a document reference
+        #   nyc_ref = firestore.doc "cities/NYC"
+        #
+        #   # Set a document
+        #   firestore.set(nyc_ref, { name: "New York City" })
+        #
+        # @example Set a document and merge all data:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Set a document
+        #   firestore.set("cities/NYC", { name: "New York City" }, merge: true)
+        #
+        # @example Set a document and merge only name:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Set a document
+        #   firestore.set("cities/NYC", { name: "New York City" },
+        #                 merge: [:name])
+        #
+        def set doc, data, merge: nil
```

Comment:
The same is true for all these methods - they should be a method on DocumentReference.